climakitae/util/utils.py

```diff
@@ -772,90 +772,105 @@ def summary_table(data: xr.Dataset) -> pd.DataFrame:
     return df
 
 
-def convert_to_local_time(data: xr.DataArray, selections) -> xr.DataArray:
+def convert_to_local_time(data: xr.DataArray | xr.Dataset) -> xr.DataArray | xr.Dataset:
     """
     Convert time dimension from UTC to local time for the grid or station.
 
     Args:
-        data (xarray.DataArray): Input data.
-        selections: DataParameters object containing selection details.
+        data (xarray.DataArray | xr.Dataset): Input data.
 
     Returns:
         xarray.DataArray: Data with converted time coordinate.
     """
 
+    # Only converting hourly data
+    if "frequency" not in data.attrs:
```

Comment:
This is another good place for a walrus operator:
```suggestion
    if not (frequency := data.attrs.get("frequency", None)):
```
Apologies for not catching this earlier

Reply:
You can cast this as a string if you need to, but I'd be surprised if its necessary.

Reply:
But be careful about casting as a string inside the `not (...)` statement because strings are "truthy" and the point is to get it to resolve to `None` and not `"None"`. Unfortunately python is weird in that `bool(None)` is `False` but `bool(str(None))` is `True`